### ISSUE TITLE
Save dialogs and other hilarity.

### ISSUE THREAD

**Issue by numist** [numist/Switch#55]
At this point in quirk complexity, it's worth considering building a hierarchical representation of window ownership from inference to wrap CG's more basic representation of windows.

Windows without titles tend to belong to an adjacent window in the window list (notable exceptions include GitHub), and windows that are logically part of another application's window hierarchy have (so far) come from applications that have non-normal activation options, like `com.apple.appkit.xpc.openAndSav`[sic].

Here's an example, using TextEdit as a foil:

```
{
    kCGWindowAlpha = "0.8500000238418579";
    kCGWindowBounds =     {
        Height = 10;
        Width = 503;
        X = 216;
        Y = 69;
    };
    kCGWindowIsOnscreen = 1;
    kCGWindowLayer = 0;
    kCGWindowMemoryUsage = 21756;
    kCGWindowNumber = 16111;
    kCGWindowOwnerName = TextEdit;
    kCGWindowOwnerPID = 34954;
    kCGWindowSharingState = 1;
    kCGWindowStoreType = 2;
},
{
    kCGWindowAlpha = 1;
    kCGWindowBounds =     {
        Height = 58;
        Width = 463;
        X = 235;
        Y = 281;
    };
    kCGWindowIsOnscreen = 1;
    kCGWindowLayer = 0;
    kCGWindowMemoryUsage = 111868;
    kCGWindowName = "";
    kCGWindowNumber = 16110;
    kCGWindowOwnerName = TextEdit;
    kCGWindowOwnerPID = 34954;
    kCGWindowSharingState = 1;
    kCGWindowStoreType = 2;
},
{
    kCGWindowAlpha = 1;
    kCGWindowBounds =     {
        Height = 319;
        Width = 489;
        X = 222;
        Y = 69;
    };
    kCGWindowIsOnscreen = 1;
    kCGWindowLayer = 0;
    kCGWindowMemoryUsage = 636156;
    kCGWindowName = Save;
    kCGWindowNumber = 16104;
    kCGWindowOwnerName = "com.apple.appkit.xpc.openAndSav";
    kCGWindowOwnerPID = 34956;
    kCGWindowSharingState = 1;
    kCGWindowStoreType = 2;
},
{
    kCGWindowAlpha = 1;
    kCGWindowBounds =     {
        Height = 319;
        Width = 490;
        X = 222;
        Y = 69;
    };
    kCGWindowIsOnscreen = 1;
    kCGWindowLayer = 0;
    kCGWindowMemoryUsage = 806740;
    kCGWindowName = Save;
    kCGWindowNumber = 16105;
    kCGWindowOwnerName = TextEdit;
    kCGWindowOwnerPID = 34954;
    kCGWindowSharingState = 1;
    kCGWindowStoreType = 2;
},
{
    kCGWindowAlpha = 1;
    kCGWindowBounds =     {
        Height = 412;
        Width = 640;
        X = 147;
        Y = 47;
    };
    kCGWindowIsOnscreen = 1;
    kCGWindowLayer = 0;
    kCGWindowMemoryUsage = 1101076;
    kCGWindowName = "Untitled.txt";
    kCGWindowNumber = 16101;
    kCGWindowOwnerName = TextEdit;
    kCGWindowOwnerPID = 34954;
    kCGWindowSharingState = 1;
    kCGWindowStoreType = 2;
},
```

![16105](https://f.cloud.github.com/assets/544011/1488287/d93b33a0-4745-11e3-80f3-04589651e4bd.png)
![16104](https://f.cloud.github.com/assets/544011/1488288/d9416e78-4745-11e3-8118-eb97e22998a7.png)
![16101](https://f.cloud.github.com/assets/544011/1488289/d9460a64-4745-11e3-9610-c3aacdca946b.png)
![16110](https://f.cloud.github.com/assets/544011/1488290/d946ebaa-4745-11e3-893f-6cf9c68171a7.png)
![16111](https://f.cloud.github.com/assets/544011/1488291/d94788a8-4745-11e3-8d98-8a89f9fc1c17.png)

Using just window name and application activation options is enough to coalesce this mess into the Untitled.txt window and the TextEdit Save window, but Dock sees those two as one discrete window so there needs to be a little more differentiation than that.

What. A. Mess.

**Comment by numist** [comment]
The new window filtering system is almost done (there is one application left [Tweetbot] that requires quirks to meet the previous filtering accuracy levels), but using window groups in the model (instead of just main windows) requires changes that reach up to the view layer to display correctly (plus I effectively burned the possibility of easy integration by porting Switch to a service architecture on the same branch, a lesson for next time), so #52 is getting fixed at the same time as this.

**Comment by numist** [comment]
Never mind. There's a drawing bug that isn't sorted out yet, but this issue has been fixed as of 5c2ae65bbdf33c10057c48cefcd5f3dbda7a367f.
